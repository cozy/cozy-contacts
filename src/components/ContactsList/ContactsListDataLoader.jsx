import React, { Component } from 'react'
import { PropTypes } from 'prop-types'

import ContactsList from './ContactsList'
import SpinnerContact from '../Components/Spinner'

class ContactsListDataLoader extends Component {
  componentDidUpdate() {
    if (this.props.hasMore) {
      this.props.fetchMore()
    }
  }
  render() {
    const { fetchStatus, hasMore, contacts } = this.props

    if (fetchStatus === 'loaded' && hasMore === false) {
      return <ContactsList contacts={contacts} />
    } else {
      return <SpinnerContact size="xxlarge" loadingType="fetching_contacts" />
    }
  }
}
ContactsListDataLoader.propTypes = {
  fetchStatus: PropTypes.string.isRequired,
  hasMore: PropTypes.bool.isRequired,
  contacts: PropTypes.array.isRequired,
  fetchMore: PropTypes.func.isRequired
}
export default ContactsListDataLoader
