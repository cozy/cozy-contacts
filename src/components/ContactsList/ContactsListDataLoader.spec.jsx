import React from 'react'
import { mount } from 'enzyme'

import AppLike from '../../tests/Applike'

import ContactsListDataLoader from './ContactsListDataLoader'
import renderer from 'react-test-renderer'
describe('ContactsListOrSpinner', () => {
  test('should display the loader', () => {
    const contact = { email: [{ address: 'johndoe@localhost' }] }
    const contactRowInstance = (
      <AppLike>
        <ContactsListDataLoader
          contacts={[contact]}
          hasMore={false}
          fetchStatus={'loading'}
          fetchMore={() => {}}
        />
      </AppLike>
    )
    const tree = renderer.create(contactRowInstance).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should display the contact List', () => {
    const contact = { email: [{ address: 'johndoe@localhost' }] }
    const contactRowInstance = (
      <AppLike>
        <ContactsListDataLoader
          contacts={[contact]}
          hasMore={false}
          fetchStatus={'loaded'}
          fetchMore={() => {}}
        />
      </AppLike>
    )
    const tree = renderer.create(contactRowInstance).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should call fetchMore only when needed', () => {
    const contact = { email: [{ address: 'johndoe@localhost' }] }
    const fetchMore = jest.fn()

    const contactRowInstance = (
      <AppLike>
        <ContactsListDataLoader
          contacts={[contact]}
          hasMore={true}
          fetchStatus={'loaded'}
          fetchMore={fetchMore}
        />
      </AppLike>
    )
    const comp = mount(contactRowInstance)
    expect(fetchMore).not.toBeCalled()
    comp.setProps({ contacts: [] })
    expect(fetchMore).toBeCalled()
    comp.setProps({ hasMore: false })
    expect(fetchMore).toBeCalled()
  })
})
