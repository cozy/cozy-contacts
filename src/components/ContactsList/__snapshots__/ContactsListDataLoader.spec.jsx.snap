// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactsListOrSpinner should display the contact List 1`] = `
<div
  className="list-wrapper"
>
  <ol
    className="list-contact"
  >
    <li>
      <div
        className="divider"
      >
        EMPTY
      </div>
      <ol
        className="sublist-contact"
      >
        <li>
          <div
            className="contact"
            onClick={[Function]}
          >
            <div
              className="contact-selection"
              onClick={[Function]}
            >
              <span
                data-input="checkbox"
              >
                <input
                  checked={false}
                  readOnly={true}
                  type="checkbox"
                />
                <label />
              </span>
            </div>
            <div
              className="contact-identity"
            >
              <div
                className="styles__c-avatar___PpDI- styles__c-avatar--small___1Y_Pv"
              >
                
                
                <svg
                  className="styles__icon___23x3R"
                  height="16"
                  style={Object {}}
                  width="16"
                >
                  <use
                    xlinkHref="#people"
                  />
                </svg>
              </div>
              <div
                className="u-ellipsis u-ml-1"
              >
                <span>
                  <span>
                    johndoe@localhost
                  </span>
                   
                </span>
              </div>
            </div>
            <div
              className="contact-email u-ellipsis"
            >
              johndoe@localhost
            </div>
            <div
              className="contact-phone u-ellipsis"
            >
              —
            </div>
            <div
              className="contact-cozyurl u-ellipsis"
            >
              —
            </div>
          </div>
        </li>
      </ol>
    </li>
  </ol>
  <div />
</div>
`;

exports[`ContactsListOrSpinner should display the loader 1`] = `
<div
  className="spinner"
  data-testid="contactSpinner"
>
  <div
    className="styles__c-spinner___1snK7 styles__c-spinner--middle___RwyII"
  >
    <svg
      className="styles__icon___23x3R styles__icon--spin___ybfC1"
      height={80}
      style={
        Object {
          "fill": "var(--primaryColor)",
        }
      }
      width={80}
    >
      <use
        xlinkHref="#spinner"
      />
    </svg>
    <p>
      Loading of your contacts, please wait.
    </p>
  </div>
</div>
`;
